### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by apanzerj**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,9 +4,13 @@
 
 ### Pipeline Actions
 
-There are three parts (called actions) to any project: Deploy, Build, Test.
+There are three parts (called actions) to any project: Deploy, Build, Test. These actions can occur in any order or not at all. If you don't need a build stage, then you don't need to include one. 
 
-### Signals
+#### What defines an action?
+
+Actions are defined by what they produce. Before creating a new action, you should be able to create a one sentence explaination of what that action produces (a signal? an artifact for something specific?). As you will see, each of the three basic actions produces something that defines it and makes it different from the other actions.
```
Comment (patrickgilbert-optimizely) [inline README.md]: "Actions are defined by a singular end goal."

An action results in a signal. Artifacts are a side effect.

File: README.md [hunk 2]
```diff
@@ -47,18 +51,18 @@ Thus, for the test action, the entrypoint is expected to be pipeline/test/test.
 
 ### Additional Logic
 
-In addition to `pipeline/test/test` there three other executables that are usable but not a hard and fast requirement. They are: 
+In addition to `pipeline/test/test` there three other executables that are usable but not required. They are: 
 
 ```
 pipeline/(action)/(action) -> Required
-pipeline/(action)/passed
+pipeline/(action)/success
```
Comment (patrickgilbert-optimizely) [inline README.md]: Why is this a good idea? Why shouldn't the implementation handle it's own failure modes? The pipeline shouldn't be saddled with this complexity.

Comment (apanzerj) [reply README.md]: discussed and you were okay.

File: README.md (hunk 1 above)
Comment (apanzerj) [reply README.md]: updated.

**Pull request merged**